src/zeusops_bot/models.py

```diff
@@ -26,3 +28,23 @@ class ConfigFile(TypedDict):
     """The reforger config file"""
 
     game: ConfigFileGameSection
+
+
+class TypedTypeAdapter[T]:
+    """A wrapper around Pydantic's TypeAdapter to make mypy happy
+
+    This class defines basically the same types as the actual TypeAdapter, but
+    somehow mypy doesn't complain when using this one.
+    """
+
+    def __init__(self, type: type[T], **kwargs) -> None:
+        """Typed init"""
+        self.adapter = TypeAdapter(type, *kwargs)
+
+    def validate_json(self, data: str | bytes | bytearray, /, **kwargs) -> T:
+        """Typed method"""
+        return self.adapter.validate_json(data, *kwargs)
```

Comment:
I'm not very happy about this workaround, but I wasn't able to make mypy accept
```py
def load_json[T](adapter: TypeAdapter[T], data: str) -> T:
```
without creating this dummy wrapper class.